contracts/ScheduledPayment.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.9;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+contract ScheduledPayment is Module {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.Bytes32Set;
+
+    event ScheduledPaymentSetup(
+        address indexed initiator,
+        address indexed owner,
+        address indexed avatar,
+        address target,
+        address config
+    );
+    event PaymentScheduled(uint256 nonce, bytes32 spHash);
+
+    event ConfigSet(address config);
+
+    address public config;
+    uint256 public nonce;
+    EnumerableSetUpgradeable.Bytes32Set private spHash;
+
+    modifier onlyAvatar() {
+        require(msg.sender == avatar, "caller is not the right avatar");
+        _;
+    }
+
+    constructor(
+        address _owner,
+        address _avatar,
+        address _target,
+        address _config
+    ) {
+        bytes memory initParams = abi.encode(_owner, _avatar, _target, _config);
+        setUp(initParams);
+    }
+
+    function setUp(bytes memory initParams) public override initializer {
+        (
+            address _owner,
+            address _avatar,
+            address _target,
+            address _config
+        ) = abi.decode(initParams, (address, address, address, address));
+        __Ownable_init();
+        require(_avatar != address(0), "Avatar can not be zero address");
+        require(_target != address(0), "Target can not be zero address");
+
+        avatar = _avatar;
+        target = _target;
+        config = _config;
+
+        transferOwnership(_owner);
+
+        emit ScheduledPaymentSetup(
```

Comment:
Minor: "ScheduledPaymentSetup" is a little ambiguous because it could also mean "A payment has been set up (i.e. scheduled)". Maybe `ScheduledPaymentModuleSetup` or something to be a little less ambiguous?

Reply:
Is it better to change the contract name as well to ScheduledPaymentModule?

Reply:
I think that makes sense yes, good idea

Reply:
updated